### PR TITLE
STM32 desig.c cleanup and Arch ARM GCC 5.3.0 bug workaround

### PR DESCRIPTION
**Pull request by urjaman**
This is a take it or close it kind of thing, but I'm just documenting what I had to do to get on with my stuff.

The archlinux arm gcc gives a "internal compiler error: in expand_expr_addr_expr_1, at expr.c:7736"
with that array version :(

### CODE REVIEW

**Comment by kuldeepdhaka** [comment]
i think, this should be merged even if the compiler do not give error. :+1:

**Comment by urjaman** [comment]
oh you should also change the header if you take this, apparently i was a bit fast with this and gcc didnt warn... (or i can do the change if wanted, but trivial.)

File: lib/stm32/desig.c [hunk 1]
```diff
@@ -24,7 +24,7 @@ uint16_t desig_get_flash_size(void)
 	return DESIG_FLASH_SIZE;
 }
 
-void desig_get_unique_id(uint32_t result[])
+void desig_get_unique_id(uint32_t *result)
 {
 	result[0] = DESIG_UNIQUE_ID2;
```
Comment (ChuckM) [inline lib/stm32/desig.c]: So why not then write this as 
```
    *result++ = DESIG_UNIQUE_ID2;
    *result = DESIG_UNIQUE_ID1;
```

**Comment by urjaman** [comment]
Changed it to be a bit more generic cleanup while working around the bug at the same time.,
comments?

**Comment by ChuckM** [comment]
Looks good to me :+1:

**Comment by fenugrec** [comment]
Also +1, the original code casting " uint8_t device_id[]" to a (uint32_t *) is evil.

**Comment by karlp** [comment]
I just want to check that the resulting serials are the same, but yes, this is probably the right thing to do :)

**Comment by karlp** [comment]
serials are the same,but the code gets bigger? wat?
```
   text	   data	    bss	    dec	    hex	filename
   6172	     24	    532	   6728	   1a48	old/usbmidi.elf
   6204	     24	    532	   6760	   1a68	new/usbmidi.elf
```

**Comment by karlp** [comment]
whatever, not going to even chase that, merged in d3fff11 thanks!

**Pull request closed without merging**